Neos.Media/Classes/Domain/Model/Adjustment/CropImageAdjustment.php

```diff
@@ -140,29 +151,90 @@ public function setY($y)
         return $this->y;
     }
 
+    /**
+     * This setter accepts strings in order to make configuration / mapping of settings easier.
+     *
+     * @param AspectRatio | string | null $aspectRatio
+     */
+    public function setAspectRatio($aspectRatio = null): void
+    {
+        if ($aspectRatio === null) {
+            $this->aspectRatioAsString = null;
+            return;
+        }
+        if (!$aspectRatio instanceof AspectRatio && !is_string($aspectRatio)) {
+            throw new \InvalidArgumentException(sprintf('Aspect ratio must be either AspectRatio or string, %s given.', gettype($aspectRatio)), 1552652570);
+        }
+        if (is_string($aspectRatio)) {
+            $aspectRatio = AspectRatio::fromString($aspectRatio);
+        }
+
+        $this->aspectRatioAsString = (string)$aspectRatio;
+        $this->x = null;
+        $this->y = null;
+        $this->width = null;
+        $this->height = null;
+    }
+
+    /**
+     * @return AspectRatio|null
+     */
+    public function getAspectRatio(): ?AspectRatio
+    {
+        return $this->aspectRatioAsString ? AspectRatio::fromString($this->aspectRatioAsString) : null;
+    }
+
+    /**
+     * Calculates the actual position and dimensions of the cropping area based on the given image
+     * dimensions and desired aspect ratio.
+     *
+     * @param int $originalWidth Width of the original image
+     * @param int $originalHeight Height of the original image
+     * @param AspectRatio $desiredAspectRatio The desired aspect ratio
+     * @return array Returns an array containing x, y, width and height
+     */
+    public static function calculateDimensionsByAspectRatio(int $originalWidth, int $originalHeight, AspectRatio $desiredAspectRatio): array
+    {
+        $newWidth = $originalWidth;
+        $newHeight = round($originalWidth / $desiredAspectRatio->getRatio());
+        $newX = 0;
+        $newY = round(($originalHeight - $newHeight) / 2);
+
+        if ($newHeight > $originalHeight) {
+            $newHeight = $originalHeight;
+            $newY = 0;
+            $newWidth = round($originalHeight * $desiredAspectRatio->getRatio());
+            $newX = round(($originalWidth - $newWidth) / 2);
+        }
+
+        return [$newX, $newY, $newWidth, $newHeight];
+    }
+
     /**
      * Check if this Adjustment can or should be applied to its ImageVariant.
      *
      * @param ImagineImageInterface $image
-     * @return boolean
+     * @return bool
      */
-    public function canBeApplied(ImagineImageInterface $image)
+    public function canBeApplied(ImagineImageInterface $image): bool
     {
-        if (
+        if ($this->aspectRatioAsString !== null) {
+            $desiredAspectRatio = AspectRatio::fromString($this->aspectRatioAsString);
+            $originalAspectRatio = new AspectRatio($image->getSize()->getWidth(), $image->getSize()->getHeight());
+            return $originalAspectRatio->getRatio() !== $desiredAspectRatio->getRatio();
```

Comment:
wait, why does the aspect ratio need to be different. What if I wanted to crop a smaller part but with the same aspect ratio?

Reply:
Ok, I wrote this comment before writing the overall conclusion above. It's pretty much clear now, jsut reinforces that it doesn't make much sense to me.